### PR TITLE
Metrics df

### PR DESCRIPTION
**Pull request by kristianeschenburg**
1. ```data_metrics``` now returns per-twitch and aggregate metrics as pandas DataFrames
2. ```excel_writer``` updated to reflect using DataFrames instead of dictionaries
3. tests updated
4. regression tests now use Apache Arrow files (.parquet) -- don't need to explicitly encode/decode as was required for JSON

### CODE REVIEW

File: Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,3074 @@
+{
```
Comment (jperson) [inline Untitled.ipynb]: Can you remove this file

**Comment by jperson** [comment]
Current version of pulse3d is 0.20.2, I think this change should bump the minor version, can you update the version in setup.py to 0.21.0

File: src/pulse3D/excel_writer.py [hunk 2]
```diff
@@ -359,58 +379,53 @@ def _write_xlsx(name: str, metadata_df, continuous_waveforms_df, data, is_optica
...
-    iter_idx,
-    dm,
-    continuous_waveforms_df,
-    wb,
-    continuous_waveforms_sheet,
-    snapshot_sheet,
-    full_sheet,
-    is_optical_recording,
-):
+def create_waveform_charts(iter_idx, dm, continuous_waveforms_df, wb, continuous_waveforms_sheet, snapshot_sheet, full_sheet, is_optical_recording):
```
Comment (jperson) [inline src/pulse3D/excel_writer.py:428]: `is_optical_recording` isn't defined anywhere, in `main` it's a parameter to `_write_xlsx` which is removed in this branch

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,6 @@
 [submodule "mantarray-magnet-finding"]
 	path = mantarray-magnet-finding
 	url = https://github.com/CuriBio/mantarray-magnet-finding.git
+[submodule "src/mantarray-magnet-finding"]
+	path = src/mantarray-magnet-finding
```
Comment (jperson) [inline .gitmodules:5]: Everything looks good, except I don't think the submodule is supposed to be in the `src` dir, the requirements.txt still has it in the root dir

Comment (kristianeschenburg) [reply .gitmodules:5]: Hm, don't know why there were two references to submodule -- I'm removing the `./src/mantarray-magnet-finding` reference.

**Review by jperson — APPROVED** [review]

**Pull request merged**